src/main/java/org/takes/facets/auth/PsByFlag.java

```diff
@@ -106,11 +107,13 @@ public PsByFlag(final String flg, final Map<Pattern, Pass> map) {
             final String value = flg.next();
             for (final Map.Entry<Pattern, Pass> ent : this.passes.entrySet()) {
                 if (ent.getKey().matcher(value).matches()) {
-                    users.add(ent.getValue().enter(req).next());
+                    final Opt<Iterator<Identity>> tmpIdentity = ent.getValue()
+                        .enter(req);
+                    users.add(tmpIdentity.get().next());
```

Comment:
@antonini I think here we should return as soon as the user is found (there can be only one, see line 105)